### ISSUE TITLE
APK Install Option Not Accessing apk.txt File Correctly

### ISSUE THREAD

**Issue by Jeffers24** [a4k-openproject/plugin.program.openwizard#94]
Kodi 18.6, Android phone, both wizards in the same Kodi build, both wizards using same apk.txt file.

When I choose APK Install option on my old wizard I get these options:-
![Screenshot_2020-04-25-06-59-17-991_org xbmc kodi](https://user-images.githubusercontent.com/47518199/80272804-8584b400-86c4-11ea-876b-b6435081fa1b.jpg)

When I choose APK install on my modified OpenWizard I get this option:-
![Screenshot_2020-04-25-07-00-31-494_org xbmc kodi](https://user-images.githubusercontent.com/47518199/80272834-e0b6a680-86c4-11ea-9d8b-8049d2abab3c.jpg)

When I select Official Kodi APK's I get this error:-
![Screenshot_2020-04-25-07-00-38-220_org xbmc kodi](https://user-images.githubusercontent.com/47518199/80272864-2d9a7d00-86c5-11ea-8594-10cec2337d48.jpg)

Any ideas on what I am doing wrong?

**Comment by drinfernoo** [comment]
How could I possibly know given that little information?

[https://github.com/drinfernoo/plugin.program.openwizard/wiki/Installing .apk's](https://github.com/drinfernoo/plugin.program.openwizard/wiki/Installing%20.apk's)

**Comment by Jeffers24** [comment]
Happy to supply any other information that would be useful to identify the issue.

What information would you like me to supply?

Here is an extract for one of the items in my apk.txt file on dropbox, all the entries in the apk.txt follow this format:-
name="CPU Monitor"
section="no"
url="https://www.dropbox.com/s/vom5v5dl8iwn3ry/CPU%20Monitor.apk?dl=1"
icon="https://www.dropbox.com/s/pol0ke6iq4g26qd/CPU%20Monitor.PNG?dl=1"
fanart="http://"
adult="No"
description="CPU Monitor"

I have double checked this format with the example you give in your wiki.

**Comment by SerpentDrago** [comment]
How are you even getting the apk menu to show up . 

I've tested with latest openwizard . adding only link to my apk txt in uservar  and menu doesn't appear .. at least in my test android emulator . 

What version of the wizard are you suing for both examples . 

can you post your uservar file , and apk text file .

**Comment by Jeffers24** [comment]
Hi,
Nothing to do with uservar file or apk text file.
**drinfernoo**   provided me this link, showing where the test was that stops the 'APK Installs' being listed in the OpenWizard menu list.

https://github.com/drinfernoo/plugin.program.openwizard/blob/master/resources/libs/gui/main_menu.py#L72

I changed the '<18' check, to '>16'

**Comment by SerpentDrago** [comment]
I'm fully aware that it showing or not is  Nothing to do with uservar file or txt file . i was asking for that  to look into YOUR issue . 

I just ran into another issue when trying to replicate yours (as you showed in the code above ) forgot about that :)

**Comment by SerpentDrago** [comment]
Was able to replicate issue , looking into it . 

This would have been super simple if you would have provided a debug log and a wizard log .

**Comment by SerpentDrago** [comment]
Quick fix , proper fix will be issued later by dr when he has the time . 

change line 160 in menu.py to this 

`            match = re.compile('name="(.+?)".+?ection="(.+?)".+?rl="(.+?)".+?con="(.+?)".+?anart="(.+?)".+?dult="(.+?)".+?escription="(.+?)"', re.S).findall(TEMPAPKFILE)`

as can be seen here 

https://github.com/drinfernoo/plugin.program.openwizard/compare/master...SerpentDrago:patch-1

Understand as i'm sure you know , you can't install apks in kodi 18 and up at all unless you are running a special forked version of kodi with the api target lowered (android security issue )

**Comment by Jeffers24** [comment]
SerpentDrago
Apologies for not understanding what you were asking me for and thanks for responding, I am new to to using OpenWizard and this github, I didn't know who you were. I thought you were someone having the same issue and asking me how I had done what I did.
I did offer to supply any information required to help investigate this issue, see below:-
![Screenshot_18](https://user-images.githubusercontent.com/47518199/80341886-20af9200-885b-11ea-9366-443808d40608.png)
I will always be happy to supply additional information on an issue, if there is a standard check list of items to be included with a bug report let me know I will make sure I supply items on the list in future. In the absence of a list, in future when reporting a bug I will always include a debug log and wizard log as you suggest.
Much appreciate your response and suggested change to get the apk file contents to display. I'll give it a go shortly and report back.
I am aware that standard Kodi will not allow apk installs. If I am using standard Kodi the apk is downloaded into the Packages folder and I can then install it using my file explorer, a two step installation.
I also have also created a Kodi 18.6 fork including the necessary mod so that my fork will install apks.
I'll get back to you after testing changes later.

**Comment by SerpentDrago** [comment]
Not a problem . Generally Screen shots are great . but we also normally request  FULL Debug log , and wizard log . they tell us so much about your setup / other possible conflicting addons / errors / and more . and any non standard edits you have done to the code base .   Its why dr closed your comment at first . there is simply not enough to go on .  And to be fair with all your kodi 19 stuff he may have just figured it was another one of those . but we had no idea without the logs . You have opened a shit load of issues lately . 

All good now . Dont' take it personally .

**Comment by Jeffers24** [comment]
Just started using OpenWizard as I was looking for a wizard to use in my Kodi 19 build.
That's why I have raised a number of issues/queries in the last week.

Just noticed version 1.3.7 has been released :-)

Regards Kodi 19 OpenWizard issues, I can leave off testing OpenWizard with Kodi 19 and just focus on Kodi 18.6 if that would be helpful?

I'll do my testing on Kodi 18.6 with the new version 1.3.7 of OpenWizard, until I hear back from you regards testing with Kodi 19.

Just going to test the APK changes with 1.3.7

**Comment by Jeffers24** [comment]
Update on installing APK testing.
Using Kodi 18.6 fork, OpenWizard 1.3.7 + two APK changes supplied in this thread.
Selected 'APK Installer' from OpenWizard top level menu.
APKs detailed in my apk.txt file were listed in the OpenWizard APK Install menu.
Selected 'CPU Monitor'.
Selected 'Download' and apk was downloaded.
Wizard screen displayed giving two options 'Cancel' and 'Install', install was highlighted.
I pressed OK on my remote, heard gui OK key sound, screen did not change.
I pressed left key on my remote to move to cancel option, heard gui key sound, screen did not change.
I pressed back key on my remote, screen changed to what I call the android install screen, the screen normally displayed when I select 'Install' in the wizard screen.
Selected 'Install' and CPU Monitor was installed.
APK install was sucessful, just issue with the Wizard install screen not changing after selecting install?
My kodi log is here:- https://paste.kodi.tv/gujudibene
[wizard.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4540455/wizard.log)

My wizard log is attached

**Comment by drinfernoo** [comment]
@Jeffers24, if you would please test the latest on `develop` for me? I've converted to using the Kodi Android Updater add-on for official APKs, adopted a similar method to that add-on in order to allow the wizard to launch a file manager for subsequent install, and also removed the version restriction for the APK menu to show.

**Comment by Jeffers24** [comment]
@drinfernoo , How do I get to the zip download version that you want me to test?
I have tested with v 1.3.7 and provided logs.
I can't find any later version on the code tab.

**Comment by Jeffers24** [comment]
@drinfernoo 
**Testing:-**
Finally found out how to get to Develop and downloaded version for testing.
Displaying 'APK Install' in OpenWizard menus worked.
Displaying list of APKs from my apk.txt worked.
Selected 'CPU Monitor' popup menu appeared selected download worked, apk downloaded to download folder.
CPU Monitor APK did not install automatically, this screen was displayed:-
![Install Downloaded APK](https://user-images.githubusercontent.com/47518199/80571983-6dc76b80-89f5-11ea-874f-3187acd8e18e.png)
I selected CPU Monitor and it installed worked.

The default file explorer on on my shield was the shield one, which caused me a problem getting out of it, as it didn't respond to the back key on my air mouse remote.
The back key on my shield remote did work.
Changed the default file explorer on my shield to FX file explorer and the back key on my air mouse remote works.

Using the latest OpenWizard from github develop, to install apk via the wizard all working fine for me, thanks for all your efforts.

**I have a change request:-**
The CPU Monitor APK was downloaded into the 'Download' folder pic. below. Would it be possible to download the APK into the 'Packages' folder in Kodi addons folder, then it would stay or be cleared depending on the OpenWizard packages settings I choose?
![Download Folder](https://user-images.githubusercontent.com/47518199/80572608-77050800-89f6-11ea-8333-93dce892b216.png)
